### PR TITLE
[14.0][MIG] l10n_ch_pain_credit_transfer

### PR DESCRIPTION
**Pull request by SimoneVagile**
Depends on:
- [ ] [https://github.com/OCA/l10n-switzerland/pull/591](https://github.com/OCA/l10n-switzerland/pull/591)

### CODE REVIEW

File: l10n_ch_pain_credit_transfer/data/pain.001.001.03.ch.02.xsd
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/data/pain.001.001.03.ch.02.xsd]: This schema is coming from an external source so please do not modify it.
If linters have any problem with that, just exclude it from files to be checked with something like https://github.com/OCA/l10n-italy/blob/00f0eb7417026f8b1a6eea3a0f8a8c299938eb74/.pre-commit-config.yaml#L17-L18

File: l10n_ch_pain_credit_transfer/__manifest__.py
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/__manifest__.py]: Please include the modifications done by linters in a separated commit as explained in https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0#technical-method-to-migrate-a-module-from-130-to-140-branch

**Review by SimoRubi — CHANGES_REQUESTED** [review]

File: l10n_ch_pain_credit_transfer/data/pain.001.001.03.ch.02.xsd
Comment (SimoneVagile) [reply l10n_ch_pain_credit_transfer/data/pain.001.001.03.ch.02.xsd]: @SimoRubi thanks

File: l10n_ch_pain_credit_transfer/__manifest__.py
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/__manifest__.py:7]: In https://github.com/OCA/l10n-switzerland/pull/591/commits/e0c42d4791eaac8dcf28963b930191349584401c#diff-d1cc26a7f91e6e352eccbb078ed1076f7a484b756be782a340199762861e23a4R8 this same modification is in the linters' commit but here it is in the migration commit.
Is this coming from linters or a manual change? Please update this or https://github.com/OCA/l10n-switzerland/pull/591 accordingly

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: Looking at the old test, the invoices here should be paid, why are you now just checking they have been posted?

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: Pleas avoid mentioning the partner's name in the variable, otherwise next time it changes we will have again a big diff for nothing.

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: Same comment as similar line's in above test

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoRubi) [inline l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: Please explicitly use Odoo's format for the date, or just a `date` object

**Review by SimoRubi — CHANGES_REQUESTED** [review]

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoneVagile) [reply l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: In `12.0` when you validate an invoice, you call `action_invoice_open` method of `account.invoice`.

In `14.0`, instead, you call `action_post` method of `account.move`: this changes field `state` from `Draft` to `Posted`.

[https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L139-L142](https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L139-L142)

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoneVagile) [reply l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: [https://github.com/OCA/l10n-switzerland/pull/593#discussion_r674862246](https://github.com/OCA/l10n-switzerland/pull/593#discussion_r674862246)

File: l10n_ch_pain_credit_transfer/__manifest__.py
Comment (SimoneVagile) [reply l10n_ch_pain_credit_transfer/__manifest__.py:7]: pre-commit doesn't change module version for itself, it warn you that is wrong. So, this is a manual change.

Has linters' commit only automathic changes? If it is, [https://github.com/OCA/l10n-switzerland/pull/591](https://github.com/OCA/l10n-switzerland/pull/591) is wrong.

Comment (SimoneVagile) [reply l10n_ch_pain_credit_transfer/__manifest__.py:7]: Changed in [https://github.com/OCA/l10n-switzerland/pull/591](https://github.com/OCA/l10n-switzerland/pull/591)

File: l10n_ch_pain_credit_transfer/tests/test_ch_sct.py
Comment (SimoRubi) [reply l10n_ch_pain_credit_transfer/tests/test_ch_sct.py]: In `12.0` this assertion checked that the invoice had been paid, in `14.0` it should still check that the invoice has been paid independently from how the single fields have changed.

Since `inv.state` does not have the value `paid` anymore, maybe you should check another field, probably `payment_state`: https://github.com/odoo/odoo/blob/487a90a14f15ea1983ef98aba419320c1f0cc4e7/addons/account/models/account_move.py#L228

**Review by SimoRubi — CHANGES_REQUESTED** [review]

**Review by SimoRubi — COMMENTED** [review]
Code review: looks good to me.

Will approve when bots are 🟢, this still has to wait for https://github.com/OCA/l10n-switzerland/pull/591

**Review by stevech091 — APPROVED** [review]
functional review ok

**Pull request closed without merging**
